runtime/onert/backend/train/ops/PadLayer.cc

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "PadLayer.h"
+
+#include <cker/train/operation/Pad.h>
+
+namespace onert
+{
+namespace backend
+{
+namespace train
+{
+namespace ops
+{
+
+PadLayer::PadLayer()
+  : _input(nullptr), _output(nullptr), _padData(), _padRank(), _constantValueData(),
+    _back_prop_input{nullptr}, _back_prop_output{nullptr}
+{
+  // DO NOTHING
+}
+
+template <typename T> void PadLayer::padImpl(const T *constant_value_data)
+{
+  // TODO: Use nnfw::cker::Pad
+  nnfw::cker::train::Pad<T>(_padData, _padRank,
+                            getShape(_input), getBuffer<T>(_input),
+                            getShape(_output), getBuffer<T>(_output),
+                            constant_value_data);
+}
+
+template <typename T> void PadLayer::depad()
+{
+  nnfw::cker::train::Depad<T>(_padData, _padRank,
+                              getShape(_back_prop_output), getBuffer<T>(_back_prop_output),
+                              getShape(_back_prop_input), getBuffer<T>(_back_prop_input));
+}
+
+void PadLayer::configure(const IPortableTensor *input, IPortableTensor *output,
+                         const int32_t *padData, int32_t padRank, const void *constantValueData,
+                         IPortableTensor *back_prop_input, const IPortableTensor *back_prop_output)
+{
+  _input = input;
+  _output = output;
+  memcpy(_padData, padData, sizeof(_padData));
+  _padRank = padRank;
+  _constantValueData.v = constantValueData;
+  _back_prop_input = back_prop_input;
+  _back_prop_output = back_prop_output;
+}
+
+void PadLayer::forward(bool)
+{
```

Comment:
How about using cker/cpu/Pad directly instead of implementing it here?

Reply:
Maybe you mentioned on #12522, right?